iDiff/utils/image_prep_utils.go

```diff
@@ -129,32 +89,47 @@ type CloudPrepper struct {
 }
 
 func (p CloudPrepper) ImageToFS() (string, error) {
-	// check client compatibility with Docker API
-	valid, err := ValidDockerVersion()
+	URLPattern := regexp.MustCompile("^.+/(.+(:.+){0,1})$")
+	URLMatch := URLPattern.FindStringSubmatch(p.Source)
+	path := strings.Replace(URLMatch[1], ":", "", -1)
```

Comment:
can you explain these three lines? this is kind of black magicky. maybe put a little comment or something